src/experimental/riff-raff-yaml-file/index.test.ts

```diff
@@ -1128,4 +1128,66 @@ describe("The RiffRaffYamlFileExperimental class", () => {
       "
     `);
   });
+
+  it("Should support user-added Riff-Raff deployments", () => {
+    const app = new App({ outdir: "/tmp/cdk.out" });
+    class MyApplicationStack extends GuStack {}
+    const { stack, region } = new MyApplicationStack(app, "App-PROD-deploy", {
+      stack: "deploy",
+      stage: "PROD",
+      env: { region: "eu-west-1" },
+    });
+
+    const riffraff = new RiffRaffYamlFileExperimental(app);
+
+    riffraff.riffRaffYaml.deployments.set("upload-my-static-files", {
+      actions: ["uploadStaticFiles"],
+      app: "my-static-site",
+      contentDirectory: "my-static-site",
+      dependencies: [],
+      parameters: {
+        bucketSsmKey: `/${stack}/my-static-site-origin`,
+        publicReadAcl: false,
+        cacheControl: "public, max-age=315360000, immutable",
+      },
+      regions: new Set([region]),
+      stacks: new Set([stack]),
+      type: "aws-s3",
+    });
+
+    const actual = riffraff.toYAML();
+
+    expect(actual).toMatchInlineSnapshot(`
+      "allowedStages:
+        - PROD
+      deployments:
+        cfn-eu-west-1-deploy-my-application-stack:
+          type: cloud-formation
+          regions:
+            - eu-west-1
+          stacks:
+            - deploy
+          app: my-application-stack
+          contentDirectory: /tmp/cdk.out
+          parameters:
+            templateStagePaths:
+              PROD: App-PROD-deploy.template.json
+        upload-my-static-files:
+          actions:
+            - uploadStaticFiles
+          app: my-static-site
+          contentDirectory: my-static-site
+          dependencies: []
```

Comment:
is `[]` a valid value in YAML?

If you pass a non-empty array for `dependencies` then you get the expected output, e.g. `dependencies: ['a', 'b']` turns into:

```yaml
dependencies:
  - a
  - b
```

If `[]` isn't valid, I'm not sure if this represents a bug in the riffraff generator, or if passing an empty array is just an unexpected use?

Reply:
Pasting this YAML into the [Riff-Raff validator tool](https://riffraff.gutools.co.uk/configuration/validation) suggests it is valid.

It's a good point though, and an example of the lack of validation. `dependencies` is an optional property, it's only appearing in the YAML because we've set above when calling `riffraff.riffRaffYaml.deployments.set`.

Reply:
From the [YAML spec](https://yaml.org/spec/1.2.2/#sequence):

> Sequence
> The content of a sequence node is an ordered series of zero or more nodes. In particular, a sequence may contain the same node more than once. It could even contain itself.

That last part is confusing!

> It could even contain itself.

WHAT?!

Reply:
> `dependencies` is an optional property, it's only appearing in the YAML because we've set above when calling `riffraff.riffRaffYaml.deployments.set`. A nice example of user error!

I've simplified the test now - https://github.com/guardian/cdk/pull/2042/commits/fc6e23f8b039ed090493a82c2403789ae9f4897a.